src/release.js

```diff
@@ -470,5 +471,66 @@ module.exports = function taoExtensionReleaseFactory(params = {}) {
 
             log.done(`${data.extension.name} is clean`);
         },
+
+        /**
+         * Select releasing branch
+         * - picking version-to-release CLI option and find branch with version on it
+         * - or find the biggest version and find branch with version on it
+         */
+        async selectReleasingBranch() {
+            // Filter all branches to the ones that have release in the name
+            await gitClient.fetch({'--prune': true});
+            const allBranches = await gitClient.getLocalBranches();
+
+            if (versionToRelease) {
+                const branchName = `remotes/${origin}/${branchPrefix}-${versionToRelease}`;
+                if (allBranches.includes(branchName)) {
+                    data.releasingBranch = branchName;
+                    data.version = versionToRelease;
+                }
+            } else {
+                const branchName = `remotes/${origin}/${branchPrefix}`;
+                const possibleBranches = allBranches.filter(branch => branch.includes(branchName));
+                const highestVersionBranch = this.getHighestVersionBranch(possibleBranches);
+                if (highestVersionBranch && highestVersionBranch.branch && highestVersionBranch.version) {
+                    data.releasingBranch = highestVersionBranch.branch;
+                    data.version = highestVersionBranch.version;
+                }
+            }
+
+            if (data.releasingBranch) {
+                log.done(`Branch ${data.releasingBranch} is selected.`);
+                return;
+            }
+
+            log.exit('Cannot find any branch with valid version.');
+        },
```

Comment:
When we exit, I would like to at least show the user the branch name we searched for.

I think this means you will need to exit in 2 places, with different messages.

first time `log.exit('Cannot find the branch remotes/${origin}/${branchPrefix}-${versionToRelease}');` after L490
second time `log.exit('Cannot find any branches matching remotes/${origin}/${branchPrefix}-');` after L498

Reply:
1st -> log.exit(`Cannot find the branch ${branchName}`);
2nd -> log.exit(`Cannot find any branches matching ${partialBranchName}`);

Reply:
great ;)